subprojects/packagefiles/soundtouch/source/SoundTouch/meson.build

```diff
@@ -16,17 +16,17 @@ soundtouch_sources = files(
 )
 
 # This library requires WINDOWS_EXPORT_ALL_SYMBOLS to export all the relevant
-# classes. That's beyond mindbending to do for C++, see how it's done:
+# classes under MSVC. That's beyond mindbending to do for C++, see how it's done:
 # https://github.com/Kitware/CMake/blob/master/Source/bindexplib.cxx#L237
-if system in ['windows', 'cygwin']
+if cpp.get_linker_id() in ['link', 'lld-link'] # MSVC relies on __declspec

```

Comment:
I think it also works with clang-cl

Reply:
No, unfortunately it's a linker level issue (no functions or classes are declspec marked): https://github.com/mesonbuild/wrapdb/pull/1456#discussion_r1530610649

For completeness:

<details><summary>cl + LINK</summary>

```
soundtouch on  meson [!?] via △ v3.27.9 
❯ meson setup build --wipe
The Meson build system
Version: 1.3.1
Source dir: E:\wraps\soundtouch     
Build dir: E:\wraps\soundtouch\build
Build type: native build
Project name: soundtouch
Project version: 2.3.2  
Activating VS 17.10.0 Preview 2.0
C compiler for the host machine: cl (msvc 19.40.33617.1 "Microsoft (R) C/C++ Optimizing Compiler Version 19.40.33617.1 for x64")
C linker for the host machine: link link 14.40.33617.1
C++ compiler for the host machine: cl (msvc 19.40.33617.1 "Microsoft (R) C/C++ Optimizing Compiler Version 19.40.33617.1 for x64")
C++ linker for the host machine: link link 14.40.33617.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Dependency openmp skipped: feature openmp disabled
Library m found: NO
Compiler for C++ supports arguments /fp:fast: YES 
Compiler for C++ supports arguments -ffast-math: NO 
Checking if "targeting Android with soft float implementation" compiles: NO 
Fetching value of define "__clang__" : (undefined) 
Compiler for C++ supports arguments -Wno-unused-variable: NO 
Compiler for C++ supports arguments -Wno-sign-compare: NO 
Compiler for C++ supports arguments -Wno-unused-but-set-variable: NO 
Compiler for C++ supports arguments -Wno-inconsistent-missing-override: NO 
Compiler for C++ supports arguments -Wno-unused-const-variable: NO 
Configuring soundtouch_config.h using configuration
Has header "afxres.h" : NO 
Windows resource compiler: Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
Did not find pkg-config by name 'pkg-config'
Found pkg-config: NO
Build targets in project: 3

soundtouch 2.3.2

  User defined options
    default_library: shared

Found ninja-1.10.2 at C:\ProgramData\chocolatey\bin\ninja.EXE

soundtouch on  meson [!?] via △ v3.27.9 took 6s
❯ meson compile -C build 
Activating VS 17.10.0 Preview 2.0
INFO: automatically activated MSVC compiler environment
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: C:\ProgramData\chocolatey\bin\ninja.EXE -C E:/wraps/soundtouch/build
ninja: Entering directory `E:/wraps/soundtouch/build'
[19/19] Linking target source/SoundTouchDLL/SoundTouchDLL.dll
FAILED: source/SoundTouchDLL/SoundTouchDLL.dll source/SoundTouchDLL/SoundTouchDLL.pdb
"link"  /MACHINE:x64 /OUT:source/SoundTouchDLL/SoundTouchDLL.dll source/SoundTouchDLL/source_SoundTouchDLL_SoundTouchDLL.rc_SoundTouchDLL.res source/SoundTouchDLL/SoundTouchDLL.dll.p/SoundTouchDLL.cpp.obj "/release" "/nologo" "/DEBUG" "/PDB:source\SoundTouchDLL\SoundTouchDLL.pdb" "/DLL" "/IMPLIB:source\SoundTouchDLL\SoundTouchDLL.lib" "source/SoundTouch/SoundTouch.lib" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
LINK : fatal error LNK1181: cannot open input file 'source\SoundTouch\SoundTouch.lib'
ninja: build stopped: subcommand failed.
```
</details>

<details><summary>clang + LINK</summary>

```
soundtouch on  meson [!?] via △ v3.27.9 
❯ meson setup build --wipe
The Meson build system
Version: 1.3.1
Source dir: E:\wraps\soundtouch
Build dir: E:\wraps\soundtouch\build
Build type: native build
Project name: soundtouch
Project version: 2.3.2
Activating VS 17.10.0 Preview 2.0
C compiler for the host machine: clang (clang 17.0.3 "clang version 17.0.3")
C linker for the host machine: clang link 14.40.33617.1
C++ compiler for the host machine: clang (clang 17.0.3 "clang version 17.0.3")
C++ linker for the host machine: clang link 14.40.33617.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Dependency openmp skipped: feature openmp disabled
Library m found: NO
Compiler for C++ supports arguments /fp:fast: NO 
Compiler for C++ supports arguments -ffast-math: YES 
Checking if "targeting Android with soft float implementation" compiles: NO 
Fetching value of define "__clang__" : 1 
Fetching value of define "__GNUC__" : (undefined) 
Compiler for C++ supports arguments -fgnuc-version=10: YES 
Compiler for C++ supports arguments -Wno-unused-variable: YES 
Compiler for C++ supports arguments -Wno-sign-compare: YES 
Compiler for C++ supports arguments -Wno-unused-but-set-variable: YES 
Compiler for C++ supports arguments -Wno-inconsistent-missing-override: YES 
Compiler for C++ supports arguments -Wno-unused-const-variable: YES 
Configuring soundtouch_config.h using configuration
Has header "afxres.h" : NO 
Windows resource compiler: Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
Did not find pkg-config by name 'pkg-config'
Found pkg-config: NO
Build targets in project: 3

soundtouch 2.3.2

  User defined options
    default_library: shared

Found ninja-1.10.2 at C:\ProgramData\chocolatey\bin\ninja.EXE

soundtouch on  meson [!?] via △ v3.27.9 took 11s
❯ meson compile -C build  
Activating VS 17.10.0 Preview 2.0
INFO: automatically activated MSVC compiler environment
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: C:\ProgramData\chocolatey\bin\ninja.EXE -C E:/wraps/soundtouch/build
ninja: Entering directory `E:/wraps/soundtouch/build'
[18/19] Compiling C++ object source/SoundTouchDLL/SoundTouchDLL.dll.p/SoundTouchDLL.cpp.obj
../source/SoundTouchDLL/SoundTouchDLL.cpp:126:5: warning: 'strncpy' is deprecated: This function or variable may be unsafe. Consider using strncpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [-Wdeprecated-declarations]
  126 |     strncpy(versionString, SoundTouch::getVersionString(), bufferSize - 1);
      |     ^
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\string.h:334:1: note: 'strncpy' has been explicitly marked deprecated here
  334 | __DEFINE_CPP_OVERLOAD_STANDARD_NFUNC_0_2_EX(
      | ^
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\corecrt.h:1935:17: note: expanded from macro '__DEFINE_CPP_OVERLOAD_STANDARD_NFUNC_0_2_EX'
 1935 |                 _CRT_INSECURE_DEPRECATE(_SecureFuncName) _DeclSpec _ReturnType __cdecl _FuncName(_SalAttributeDst _DstType *_Dst, _TType1 _TArg1, _TType2 _TArg2);
      |                 ^
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\vcruntime.h:355:55: note: expanded from macro '_CRT_INSECURE_DEPRECATE'
  355 |         #define _CRT_INSECURE_DEPRECATE(_Replacement) _CRT_DEPRECATE_TEXT(    \
      |                                                       ^
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\vcruntime.h:345:47: note: expanded from macro '_CRT_DEPRECATE_TEXT'
  345 | #define _CRT_DEPRECATE_TEXT(_Text) __declspec(deprecated(_Text))
      |                                               ^
1 warning generated.
[19/19] Linking target source/SoundTouchDLL/SoundTouchDLL.dll
FAILED: source/SoundTouchDLL/SoundTouchDLL.dll source/SoundTouchDLL/SoundTouchDLL.pdb
"clang"  -Wl,/MACHINE:X64 -Wl,/OUT:source/SoundTouchDLL/SoundTouchDLL.dll source/SoundTouchDLL/source_SoundTouchDLL_SoundTouchDLL.rc_SoundTouchDLL.res source/SoundTouchDLL/SoundTouchDLL.dll.p/SoundTouchDLL.cpp.obj "-Wl,/release" "-Wl,/nologo" "-Wl,/DEBUG" "-Wl,/PDB:source\SoundTouchDLL\SoundTouchDLL.pdb" "-Wl,/DLL" "-Wl,/IMPLIB:source\SoundTouchDLL\SoundTouchDLL.lib" "source/SoundTouch/SoundTouch.lib" "-lkernel32" "-luser32" "-lgdi32" "-lwinspool" "-lshell32" "-lole32" "-loleaut32" "-luuid" "-lcomdlg32" "-ladvapi32"
clang: error: no such file or directory: 'source/SoundTouch/SoundTouch.lib'
ninja: build stopped: subcommand failed.

```
</details>